api/checks/update.go

```diff
@@ -0,0 +1,148 @@
+// Copyright 2018 The WPT Dashboard Project. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+package checks
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+
+	"github.com/deckarep/golang-set"
+
+	"github.com/gorilla/mux"
+	"github.com/web-platform-tests/wpt.fyi/api/checks/summaries"
+	"github.com/web-platform-tests/wpt.fyi/shared"
+	"google.golang.org/appengine"
+)
+
+// updateCheckHandler handles /api/checks/[commit] POST requests.
+func updateCheckHandler(w http.ResponseWriter, r *http.Request) {
+	vars := mux.Vars(r)
+	sha := vars["commit"]
+	if len(sha) != 40 {
+		http.Error(w, fmt.Sprintf("Invalid commit: %s", sha), http.StatusBadRequest)
```

Comment:
So one tip about designing the endpoint of TaskQueue: if the error can't be recovered by retrying (e.g. invalid parameters), you'd need to return a 2XX status code to "pretend" a success otherwise the task will be retried forever and gets piled up.

Reply:
We can also explicitly set a reasonable `task_retry_limit`, e.g. 3?